core/src/main/java/org/hisp/dhis/android/core/arch/call/factories/internal/QueryCoroutineCallFactoryImpl.kt

```diff
@@ -25,34 +25,23 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-
-package org.hisp.dhis.android.core.arch.call.factories.internal;
-
-import org.hisp.dhis.android.core.arch.api.executors.internal.APICallExecutor;
-import org.hisp.dhis.android.core.arch.call.fetchers.internal.CallFetcher;
-import org.hisp.dhis.android.core.arch.call.internal.EndpointCall;
-import org.hisp.dhis.android.core.arch.call.internal.GenericCallData;
-import org.hisp.dhis.android.core.arch.call.processors.internal.CallProcessor;
-import org.hisp.dhis.android.core.arch.call.queries.internal.BaseQuery;
-
-import java.util.List;
-import java.util.concurrent.Callable;
-
-public abstract class QueryCallFactoryImpl<P, Q extends BaseQuery> implements QueryCallFactory<P, Q> {
-
-    protected final GenericCallData data;
-    protected final APICallExecutor apiCallExecutor;
-
-    protected QueryCallFactoryImpl(GenericCallData data, APICallExecutor apiCallExecutor) {
-        this.data = data;
-        this.apiCallExecutor = apiCallExecutor;
-    }
-
-    @Override
-    public final Callable<List<P>> create(Q query) {
-        return new EndpointCall<>(fetcher(query), processor(query));
+package org.hisp.dhis.android.core.arch.call.factories.internal
+
+import org.hisp.dhis.android.core.arch.api.executors.internal.CoroutineAPICallExecutor
+import org.hisp.dhis.android.core.arch.call.internal.GenericCallData
+import org.hisp.dhis.android.core.arch.call.processors.internal.CallProcessor
+import org.hisp.dhis.android.core.arch.call.queries.internal.BaseQueryKt
+
+internal abstract class QueryCoroutineCallFactoryImpl<P, Q : BaseQueryKt> protected constructor(
+    protected val data: GenericCallData,
+    protected val coroutineAPICallExecutor: CoroutineAPICallExecutor,
+) : QueryCallFactory<P, Q> {
+    override suspend fun create(query: Q): List<P> {
+        val objects: List<P> = fetcher(query)
+        processor(query).process(objects)
+        return objects
     }
 
-    protected abstract CallFetcher<P> fetcher(Q query);
-    protected abstract CallProcessor<P> processor(Q query);
-}
\ No newline at end of file
+    protected abstract suspend fun fetcher(query: Q): List<P>
```

Comment:
Following the convention, this would be the `CoroutineCallFetcher` you created in the other PR. In the `create()` method, it will call to `fetcher(query).fetch()`.